__mocks__/vscode.ts

```diff
@@ -255,3 +256,18 @@ export class CodeLens {
     public command?: Command
   ) { }
 }
+
+export const authentication = {
+  getSession: vi.fn()
+}
+
+export class WorkspaceEdit {
+  set() {
+    
+  }
+}
+
+export class NotebookEdit {
+  static updateCellMetadata = vi.fn()
+
```

Comment:
```suggestion
```

Reply:
> Works and looks great!
> 
> 1. I canceled my workflow half-way through but the UI displayed it as successfully completed. Something we need to fix please.
> 2. Any idea why the file continuously shows the unsaved dot, even after saving?
> 
> <img alt="image" width="1724" src="https://user-images.githubusercontent.com/250527/239087639-6073e4d5-14c2-4d1c-9903-69c5f1775bb3.png">

@sourishkrout :
1. Fixed
2. I'm checking this issue, let me know if this is a blocker for merging this.

Reply:
@degrammer re 2., I think we need to understand what the problem is. I'm uncomfortable merging since it feels like a "bug in the making" to me. Sorry about the delay, I overlooked this ping in the noise of everything else.